src/pycell.rs

```diff
@@ -975,6 +983,7 @@ impl From<PyBorrowMutError> for PyErr {
 #[doc(hidden)]
 pub trait PyCellLayout<T>: PyLayout<T> {
     fn ensure_threadsafe(&self);
+    fn check_threadsafe(&self) -> Result<(), PyBorrowError>;
```

Comment:
Potentially this should return `bool` to match the `PyClassThreadChecker` trait implementation? It would make `try_borrow_threadsafe` implementation a bit messier but might be nice for consistency.

Reply:
I had written down both and ended up with this version because the calling (i.e. `try_borrow_threadsafe`) was more consistent this way without the `PyCellLayout` impl becoming more complicated. The other way around, both impls looked messier. But to be honest, the differences were small/purely aesthetic either way, so If you'd like me to, I can certainly change this back.

Reply:
I trust your taste for good design, so let's keep it like this then 👍